### ISSUE TITLE
Using Guzzle client to download files

### ISSUE THREAD

**Issue by bencroker** [craftcms/feed-me#648]
### Description

While migrating entries that had related assets stored in an Amazon S3 bucket, the plugin was erroring each time it tried to download a filename that had non-ascii characters in it (ä, ö, ü, ß, ...). After a lot of debugging and head-scratching I finally put it down to the download method, but neither the `CURL` nor `fopen` methods were working. I replaced them with a Guzzle request and it worked perfectly.

I've forked the plugin and am linking to the chages for reference in case this is of use to anyone else or @pixelandtonic decide to use the Guzzle client in core:
https://github.com/craftcms/feed-me/commit/8eb317c48bc6ca9fbe22e0f44e0c2b89251d99fe

### Steps to reproduce

Run a feed import with assets stored in an Amazon S3 bucket with filenames that contain non-ascii characters (ä, ö, ü, ß, ...). 

### Additional info

- Plugin version: 4.2.0
- Craft version: 3.4.0-beta5

**Comment by therealpecus** [comment]
Hey @bencroker,
I tried your suggestion while troubleshooting a similar issue (eventually unrelated).
You probably forgot to declare `$response` in your code:

https://github.com/bencroker/feed-me/blob/8eb317c48bc6ca9fbe22e0f44e0c2b89251d99fe/src/helpers/AssetHelper.php#L24

**Comment by bencroker** [comment]
It looks like I did, good catch @therealpecus, but I got the Guzzle approach to work correctly in the end. Was your issue also to do with non-ascii characters in the URIs?

**Comment by therealpecus** [comment]
no, in the end it was not related to downloading of remote assets but with their creation. I'm tracking it in #790
